includes/polling/sensors/state/microsemitime.inc.php

```diff
@@ -0,0 +1,19 @@
+<?php
+
+// Workaround for Microsemi/Symmetricom SyncServer SNMP behaviour
+// Several values that should be returned as integers are returned as strings,
+// sometimes with extra text prepended.
+if ($class == 'state') {
```

Comment:
you should change this to
if ($device['os'] == 'microsemitime') {

Reply:
Sensors only loads if the os and class matches, I don't think any if statement is needed here...

Reply:
Also, you sure the won't work with yaml?  the state sensor code tries to convert strings to values.

Reply:
Apologies for the delay in replying here. @murrant you're right, it's not necessary at all since this should only be run if the OS and sensor type already match.

That said, taking another look at the other files in this folder, what @Jellyfrog has suggested appears to be convention; save for dhcpatriot.inc.php, they all check for an OS match, I guess as a final sanity check?

It does sort of work with yaml, but the crappy SNMP MIB that they've implemented makes the result it spits out kind of cryptic, e.g.:

```
gpsHealth OBJECT-TYPE
    SYNTAX		DisplayString (SIZE(1..80))
    MAX-ACCESS	read-only
    STATUS		current	
    DESCRIPTION 
        "This is the GPS� receiver health status defined as
        follows:
        ======================================================
        0 = Receiver Down       The Hardware Clock can't 
                                communicate with the receiver.

        1 = Unknown Mode        An undefined mode of the GPS
                                receiver.

        2 = Acquiring Signal    The receiver is attempting to
                                track a GPS signal.

        3 = Bad Geometry        The geometry of the tracked
                                satellites is unsatisfactory for
                                a position solution.

        4 = Propagate Mode      A position estimation mode used
                                in highly dynamic environments.

        5 = 2d Solution         The receiver is able to perform 
                                position fixes for latitude and 
                                longitude but does not have 
                                enough satellites for altitude.

        6 = 3d Solution         The receiver is now able to 
                                perform position fixes for 
                                latitude, longitude and altitude.

        7 = Position Hold       Position fixes are no longer 
                                attempted, and the user entered 
                                or surveyed position is used.

        8 = Time Valid          The receiver has valid timing 
                                information from GPS satellites 
                                (including current leap second 
                                information).  This is the final 
                                state for all configured GPS modes.
        ======================================================"
```

This is a DisplayString, and (for example) returns `Receiver Health: 2`. Yes, I can match on that description with YAML, but since the same YAML field is used to define what string to match on as well as what string to display, I'm unable to map the returned value to its actual description (in this case, `Acquiring Signal`) for display in the web interface,

The same applies to tymingSource, the only difference being that they don't prepend another string to that one's return value; likewise, it's much more useful to display `External 1PPS` than `16`; It would be nice if the YAML state sensor format allowed to match on one string and display another.

If there's a better way to handle this (or a way to handle it solely using YAML) I'm quite happy to change my approach, but this seems to be what's been done for a few other devices with similar flaws.

Reply:
In this case the code also extracts the first integer seen in a string.  So, you should still just be able to use a regular state sensor.

Reply:
H-uh. Well. Would you look at that, so it does. Let me go clear this file out, then...